pkg/console/subresource/consoleserver/types.go

```diff
@@ -39,9 +39,17 @@ type ServingInfo struct {
 
 // ClusterInfo holds information the about the cluster such as master public URL and console public URL.
 type ClusterInfo struct {
-	ConsoleBaseAddress string `yaml:"consoleBaseAddress,omitempty"`
-	ConsoleBasePath    string `yaml:"consoleBasePath,omitempty"`
-	MasterPublicURL    string `yaml:"masterPublicURL,omitempty"`
+	ConsoleBaseAddress string           `yaml:"consoleBaseAddress,omitempty"`
+	ConsoleBasePath    string           `yaml:"consoleBasePath,omitempty"`
+	MasterPublicURL    string           `yaml:"masterPublicURL,omitempty"`
+	CLIDownloadURLs    *CLIDownloadURLs `yaml:"cliDownloadURLs,omitempty"`
+}
+
+// CLIDownloadURLs contains download URLs for each of the platforms we provide CLI binary.
+type CLIDownloadURLs struct {
```

Comment:
Maybe simplify this a bit:

```golang
type CLIDownloadURLs struct {
	Linux   string `yaml:"linuxDownloadURL"`
	Mac    string `yaml:"macDownloadURL"`
	Windows string `yaml:"windowsDownloadURL"`
```
Since its already a heading `CLIDownloadURLs` we don't need to repeat the `DownloadURL` part in the field names.

Reply:
To be honest, we may want to do CLIDownloadURLs[arch][platform] just like the downloads deployment. Only amd64 now.

Reply:
@benjaminapetersen how about? 

```go
type ClusterInfo struct {
	ConsoleBaseAddress string          `yaml:"consoleBaseAddress,omitempty"`
	ConsoleBasePath    string          `yaml:"consoleBasePath,omitempty"`
	MasterPublicURL    string          `yaml:"masterPublicURL,omitempty"`
	CLIDownloadURLs    CLIDownloadURLs `yaml:"cliDownloadURLs,omitempty"`
}

// CLIDownloadURLs contains all the architertures that we are building CLI binaries for.
type CLIDownloadURLs struct {
	Amd64 ArchPlatformsURL `yaml:"amd64,omitempty"`
}

// ArchPlatformsURL contains URLs for each of the platforms we provide CLI binary.
type ArchPlatformsURL struct {
	Linux   string `yaml:"linux"`
	Mac     string `yaml:"mac"`
	Windows string `yaml:"windows"`
}
```

Although Im not really sure about the `CLIDownloadURLs`, but its pretty generic and with the aditional architecture specified it kinda dioes make sense?

Reply:
Yup!  Though I wonder if `ArchPlatformsURL` vs `ArchPlatformURLs`.  As "one platform has many urls", and if you had many archs, you would `[]ArchPlatformURLs`,

```golang
type CLIDownloadURLs struct {
   Amd64 ArchPlatformURLs `yaml:"amd64,omitempty"`
   Foo32 ArchPlatformURLs `yaml:"Foo32,omitempty"`
   Bar64 ArchPlatformURLs `yaml:"Bar64,omitempty"`
}
```
Since we expect 3+ URLs for archs added.

Reply:
But if we tan to chain it like you mentioned in your previous comment it cant be an array.

Reply:
+1 on `ArchPlatformURLs`